build-common/src/org/jetbrains/kotlin/incremental/IncrementalJvmCache.kt

```diff
@@ -726,61 +758,66 @@ class KotlinClassInfo constructor(
         }
 
         fun createFrom(classId: ClassId, classHeader: KotlinClassHeader, classContents: ByteArray): KotlinClassInfo {
-            val constantsAndInlineFunctions = getConstantsAndInlineFunctions(classHeader, classContents)
+            val (constants, inlineFunctionsAndAccessors) = getConstantsAndInlineFunctionsOrAccessors(classHeader, classContents)
 
             return KotlinClassInfo(
                 classId,
                 classHeader.kind,
-                classHeader.data ?: classHeader.incompatibleData ?: emptyArray(),
+                classHeader.data ?: emptyArray(),
                 classHeader.strings ?: emptyArray(),
                 classHeader.multifileClassName,
-                constantsMap = constantsAndInlineFunctions.first,
-                inlineFunctionsMap = constantsAndInlineFunctions.second
+                constants.mapKeysTo(LinkedHashMap()) { it.key.name },
+                inlineFunctionsAndAccessors.mapKeysTo(LinkedHashMap()) { it.key.asString() },
             )
         }
     }
 }
 
-/** Parses the class file only once to get both constants and inline functions. */
-private fun getConstantsAndInlineFunctions(
+/**
+ * Parses the class file only once to get both constants and inline functions/property accessors. This is faster than getting them
+ * separately in two passes.
+ */
+private fun getConstantsAndInlineFunctionsOrAccessors(
     classHeader: KotlinClassHeader,
     classContents: ByteArray
-): Pair<LinkedHashMap<String, Any>, LinkedHashMap<String, Long>> {
+): Pair<Map<JvmMemberSignature.Field, Any>, Map<JvmMemberSignature.Method, Long>> {
     val constantsClassVisitor = ConstantsClassVisitor()
-    val inlineFunctionSignatures = inlineFunctionsJvmNames(classHeader)
+    val inlineFunctionsAndAccessors = inlineFunctionsAndAccessors(classHeader)
 
-    return if (inlineFunctionSignatures.isEmpty()) {
+    return if (inlineFunctionsAndAccessors.isEmpty()) {
         ClassReader(classContents).accept(constantsClassVisitor, SKIP_CODE or SKIP_DEBUG or SKIP_FRAMES)
-        Pair(constantsClassVisitor.getResult(), LinkedHashMap())
+        Pair(constantsClassVisitor.getResult(), emptyMap())
     } else {
-        val inlineFunctionsClassVisitor = InlineFunctionsClassVisitor(inlineFunctionSignatures, constantsClassVisitor)
-        ClassReader(classContents).accept(inlineFunctionsClassVisitor, 0)
-        Pair(constantsClassVisitor.getResult(), inlineFunctionsClassVisitor.getResult())
+        val inlineFunctionsAndAccessorsClassVisitor =
+            InlineFunctionsAndAccessorsClassVisitor(inlineFunctionsAndAccessors.toSet(), constantsClassVisitor)
+        // SKIP_DEBUG so that snapshotting the methods is not sensitive to line number changes
```

Comment:
Can this impact debugging of the subproject that uses this inline function? E.g. can we end up with incorrect line info because it was not updated in the consuming compilation.

Reply:
That's a fantastic point, thank you for catching this!

I've checked that the line number info is used when compiling the consuming class (as it should be), so we can't have `SKIP_DEBUG` here. Let me add a test to safeguard this case.

Reply:
Done - I've removed `SKIP_DEBUG` and added a test.